pkg/pmem-csi-driver/controllerserver-node.go

```diff
@@ -74,6 +74,21 @@ func NewNodeControllerServer(nodeID string, dm pmdmanager.PmemDeviceManager, sm
 		}
 
 		for _, id := range ids {
+			// retrieve volume info
+			vol := &nodeVolume{}
+			if err := sm.Get(id, vol); err != nil {
+				klog.Warningf("Failed to retrieve volume info for id %q from state: %v", id, err)
+				continue
+			}
+
+			v, err := parameters.Parse(parameters.NodeVolumeOrigin, vol.Params)
+			if err != nil {
+				klog.Warningf("Failed to parse volume parameters for volume %q: %v", id, err)
```

Comment:
It should also ignore volumes where it cannot parse the parameters, i.e. there's a "continue" (or something similar, see other comment) missing here. This will become relevant in the following scenario:
- create a volume with new parameters using a newer PMEM-CSI release
- downgrade to an older release which doesn't know about that parameter

The older release should not act upon that volume because the parameter might be crucial for correctly working with it.

BTW, this is a technical reason why extending parameters is better than a new field: the JSON parser will ignore unknown fields in JSON and then there is no way for this code here to detect that it doesn't have all information about the volume.